runtime/src/storage/iouring.rs

```diff
@@ -21,17 +26,131 @@ impl Config {
     }
 }
 
+// New type to handle io_uring operations
+struct IoUringRuntimeAdapter {
+    // The shared io_uring instance
+    ring: Mutex<IoUring>,
+    // Notification mechanism for completion events
+    work_available: tokio::sync::Notify,
```

Comment:
We should be able to get away from this using `futures::mpsc` (it handles all of this notification stuff automatically)

Reply:
This will also mean we can get rid of any mutex